packages/web3-types/src/eth_types.ts

```diff
@@ -45,8 +45,11 @@ export enum BlockTags {
 	EARLIEST = 'earliest',
 	LATEST = 'latest',
 	PENDING = 'pending',
+	SAFE = 'safe',
+	FINALIZED = 'finalized',
 }
-export type BlockTag = 'earliest' | 'latest' | 'pending';
+export type BlockTag = `${BlockTags}`;
+
 export type BlockNumberOrTag = Numbers | BlockTag;
```

Comment:
Since `Numbers` could be any string. I suggest we use this:
```suggestion
export type BlockNumberOrTag = number | bigint | BlockTag;
```

Reply:
Numbers already includes bigint https://github.com/web3/web3.js/blob/f3b8ca88eb34dbf73f8e3dc9f5995c4c1a54fa87/packages/web3-types/src/primitives_types.ts#L20 so this change is not required.

Reply:
Since BlockNumberOrTag is used in many other places of lib and uses can pass hex string to other functions so this shouldnt be changed.